File: src/main/scala/org/cloudio/morpheus/mail/traditional/RegisteredUserMail.scala
package org.cloudio.morpheus.mail.traditional

import java.util.Date


/**
 * Created by zslajchrt on 24/08/15.
 */
trait RegisteredUserMail extends UserMail {
  this: RegisteredUser =>

  abstract override def validateEmail(message: Message) {
    val now: Date = new Date
    if (validTo.compareTo(now) < 0) {
      throw new IllegalArgumentException("User's account expired")
    }
    super.validateEmail(message)
  }
}

File: src/main/java/org/cloudio/morpheus/mail/RegisteredUserMail.java
package org.cloudio.morpheus.mail;

import java.util.Date;
import java.util.List;

/**
 * Created by zslajchrt on 24/08/15.
 */
public class RegisteredUserMail extends DefaultUserMail {

    public RegisteredUserMail(RegisteredUserAdapter registeredUser) {
        super(registeredUser);
    }

    @Override
    public void validateEmail(Message message) {
        Date validTo = ((RegisteredUserAdapter) getMailOwner()).getValidTo();
        Date now = new Date();
        if (validTo.compareTo(now) < 0) {
            throw new IllegalArgumentException("User's account expired");
        }
        super.validateEmail(message);
    }
}
